sfm_pc/views.py

```diff
@@ -295,13 +298,20 @@ def osm_autocomplete(request):
           ST_Y(ST_Centroid(geometry)) AS latitude,
           ST_asgeojson(ST_Simplify(geometry, 0.01))::json AS geojson
         FROM osm_data
-        WHERE plainto_tsquery('english', %s) @@ search_index
     '''
 
+    if search_by_id:
+        q_args[0] += '%%'
```

Comment:
I'd appreciate a tiny tiny comment about this addition to `q_args`.

Reply:
Wise! I'll drop one in.

The gist is that the syntax for `beginswith`-like queries in SQL is:

```sql
SELECT * FROM table WHERE column LIKE "string%"
```

Where `%` represents a wildcard match. (It's doubled in the code, like `%%`, since you have to escape `%` to use them as literal characters in Python.) Unfortunately, we can't just code the wildcard portion straight into the query string on [line 305](https://github.com/security-force-monitor/sfm-cms/pull/217/files/2d3ac8f0bffbc29a861de1cf07741f3a39373fc2#diff-5fca6212fdfe86e75a7631d678505231R305); as a security measure, Django's direct DB access API (the one that uses the `connection.cursor()` object) requires that all user-submitted text be escaped in quotes before getting formatted into a given query, and the extra layer of quotes ruins the wildcard syntax:

```python
# Trying to do a query like this:
query = ''' SELECT * FROM table WHERE column LIKE "%s%%" ''' % (query_string)
```
```sql
# Would return SQL that looks this, and which throws a ProgrammingError due to the extra set of quotes:
''' SELECT * FROM table WHERE column LIKE "'query_string'%" '''
```

Obviously my comment will be much shorter than this... but I thought it was an interesting syntax problem!
```